app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt

```diff
@@ -32,15 +41,61 @@ class HomeFragment : Fragment() {
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        return inflater.inflate(R.layout.fragment_home, container, false)
+        _binding = FragmentHomeBinding.inflate(inflater, container, false)
+        val view = binding.root
+        return view
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        ViewCompat.setOnApplyWindowInsetsListener(binding.headerContent) { v, insets ->
+            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
+            v.setPadding(0, systemBars.top, 0, 0)
+            insets
+        }
+
+        adapter = HomeAdapter()
 
-        viewModel.getLandmarkList().observe(viewLifecycleOwner) { landmarkList ->
-            Log.i("MYTAG", landmarkList.toString())
+        with(binding) {
+            rvLandmark.layoutManager = LinearLayoutManager(requireContext())
+            rvLandmark.setHasFixedSize(true)
         }
 
+        viewModel.getLandmarkList().observe(viewLifecycleOwner) {
+            when (it) {
+                is HomeViewModel.LandmarkListState.Loading -> {
+                    setLoading(true)
+                }
+
+                is HomeViewModel.LandmarkListState.Result -> {
+                    setLoading(false)
+
+                    val landmarkList = it.landmarkList
+
+                    adapter.submitList(landmarkList)
+                    binding.rvLandmark.adapter = adapter
+
+                    adapter.setOnItemClickListener(object : HomeAdapter.OnItemClickListener {
+                        override fun onItemClick(landmark: Landmark) {
+                            Intent(requireContext(), LandmarkDetailActivity::class.java).apply {
+                                putExtra(LandmarkDetailActivity.EXTRA_LANDMARK_ID, landmark.id)
+                            }.run {
+                                startActivity(this)
+                            }
+                        }
+                    })
+
+                    Log.d("MYTAG", landmarkList.toString())
+                }
+            }
+        }
+    }
+
+    private fun setLoading(state: Boolean) {
+        if (state) {
+            binding.progressIndicator.visibility = View.VISIBLE
+        } else {
+            binding.progressIndicator.visibility = View.GONE
+        }
```

Comment:
for the simple mode use this one

binding.progressBar.visibility = if (isLoading) View.VISIBLE else View.GONE